module/VuFind/src/VuFind/RecordDriver/SolrDefault.php

```diff
@@ -298,4 +302,74 @@ public function getContainerRecordID()
             && !empty($this->fields['hierarchy_parent_id'])
             ? $this->fields['hierarchy_parent_id'][0] : '';
     }
+
+    /**
+     * Get work identification keys
+     *
+     * @return array
+     */
+    public function getWorkKeys()
+    {
+        return $this->fields['work_keys_str_mv'] ?? [];
+    }
+
+    /**
+     * Return count of other versions available
+     *
+     * @return int
+     */
+    public function getOtherVersionCount()
```

Comment:
Since getVersions and getOtherVersionsCount() rely on a feature interface in the search backend rather than Solr-specific logic, do you think it would be worth creating a VersionAwareTrait and VersionAwareInterface and move some of this out of SolrDefault? Then SolrDefault can define getWorkKeys (which can be required by VersionAwareInterface, and which contains a Solr-specific detail in the field name) but everything else can come in through VersionAwareTrait. I don't know how likely we are to implement this for other backends, but anything that keeps the record drivers from becoming even more monolithic may be helpful. :-)

Reply:
Good idea. I'll see what I can do.

Reply:
Perhaps it makes sense to add a VuFind\RecordDriver\Feature namespace sort of parallel to the search backend approach, so the traits are more visibly separated from the drivers themselves. We may eventually want to move the existing traits into this namespace, but maybe now would be a good time to establish it. What do you think?

Reply:
Done, hopefully in the way you envisioned.

Reply:
Looks good, thanks!